Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs

```diff
@@ -91,23 +91,61 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
             return isTestLinkExist;
         }
 
-        public async Task AddAnswerAsync(int attendeeId, string answers)
+        public async Task AddAnswerAsync(int attendeeId, TestAnswerAC answer)
         {
             if (await _dbContext.AttendeeAnswers.AnyAsync(x => x.Id == attendeeId))
             {
-                var answersToUpdate = await _dbContext.AttendeeAnswers.FindAsync(attendeeId);
-                answersToUpdate.Answers = answers;
+                var attendeeAnswer = await _dbContext.AttendeeAnswers.FindAsync(attendeeId);
+                var deserializedAnswer = new List<TestAnswerAC>();
+
+                if (attendeeAnswer.Answers != null)
+                {
+                    deserializedAnswer = JsonConvert.DeserializeObject<TestAnswerAC[]>(attendeeAnswer.Answers).ToList();
+                }
+
+                //Remove answer if already exist
+                var answerToUpdate = deserializedAnswer.SingleOrDefault(x => x.QuestionId == answer.QuestionId);
+                if (answerToUpdate != null)
+                {
+                    deserializedAnswer.Remove(answerToUpdate);
+                }
+                
+                //Add answer
+                deserializedAnswer.Add(answer);
+                var serializedAnswer = JsonConvert.SerializeObject(deserializedAnswer);
+                attendeeAnswer.Answers = serializedAnswer;
+                _dbContext.AttendeeAnswers.Update(attendeeAnswer);
             }
             else
             {
                 var attendeeAnswers = new AttendeeAnswers();
                 attendeeAnswers.Id = attendeeId;
-                attendeeAnswers.Answers = answers;
+
+                if (answer != null)
+                {
+                    var testAnswerArray = new List<TestAnswerAC>();
+                    testAnswerArray.Add(answer);
+                    attendeeAnswers.Answers = JsonConvert.SerializeObject(testAnswerArray);
+                }
                 await _dbContext.AddAsync(attendeeAnswers);
             }
             await _dbContext.SaveChangesAsync();
         }
 
+        public async Task<bool> IsAnswerValidAsync(int attendeeId, TestAnswerAC answer)
+        {
+            var attendeeAnswer = await _dbContext.AttendeeAnswers.FindAsync(attendeeId);
+            if(attendeeAnswer.Answers != null)
+            {
+                var deserializedAnswer = JsonConvert.DeserializeObject<TestAnswerAC[]>(attendeeAnswer.Answers);
+                if(deserializedAnswer[0] != null)
+                {
+                    return !(deserializedAnswer.Where(x => x.QuestionId == answer.QuestionId && x.QuestionStatus == QuestionStatus.answered).Count() > 0);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use the opposite operator ('<=') instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1940)